src/number.cr

```diff
@@ -92,6 +99,14 @@ struct Number
   # ints = Int64.slice(1, 2, 3)
   # ints.class # => Slice(Int64)
   # ```
+  #
+  # This is a convenvenient alternative to `Slice.[]` for designating a
+  # specific item type which also considers autocasting.
```

Comment:
The macro uses `new!`, which is not the same as autocasting:

```crystal
Int8.slice(0x1234, 1023.0) # => Slice[52, -1]
```

Don't know if this is intentional, but at least this doc doesn't reflect the reality. Same for `.static_array`.

Reply:
Oooh. I suppose it's probably a performance measure. But it can easily end up as a foot gun.

I'm thinking it should probably better use `new` for safety.
And if `num` is a number literal, we can use that directly for performance reasons. Implicit autocasting can handle that.

Reply:
I created a new issue for that: https://github.com/crystal-lang/crystal/issues/13033